File: 프로그래머스/0/120882. 등수 매기기/등수 매기기.py
def solution(score):
    mean_score = []
    for i in range(len(score)):
        mean_score.append(sum(score[i])/len(score[i]))
    rank_list = sorted(mean_score, reverse = True)
    answer = []
    for i in range(len(score)):
        answer.append(rank_list.index(mean_score[i])+1)
    return answer

File: 프로그래머스/0/120882. 등수 매기기/등수 매기기.java
import java.util.Arrays;
import java.util.Collections;
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] solution(int[][] score) {
        int[] answer = new int[score.length];
        Double[] averages = new Double[score.length];
        for (int i = 0; i < score.length; i++) {
            averages[i] = (score[i][0] + score[i][1]) / 2.0;
        }
        Double[] sorted = averages.clone();
        Arrays.sort(sorted, Collections.reverseOrder());
        for (int i = 0; i < score.length; i++) {
            for (int j = 0; j < score.length; j++) {
                if (averages[i].equals(sorted[j])) {
                    answer[i] = j + 1;
                    break;
                }
            }
        }
        return answer;
    }
}